### PR TITLE
direct invocation of PartitionedConfig

### PR DESCRIPTION
**Pull request by sryza**
## Checklist

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: python_modules/dagster/dagster/core/definitions/partition.py [hunk 1]
```diff
@@ -690,6 +692,15 @@ def get_run_config(self, partition_key: str) -> Dict[str, Any]:
...
         return self.run_config_for_partition_fn(matching[0])
 
+    def __call__(self, *args, **kwargs):
+        if self._decorated_fn is None:
+            raise DagsterInvalidInvocationError(
+                "Only PartitionedConfig objects created using one of the partitioned config "
+                "decorators can be directly invoked."
+            )
+        else:
+            return self._decorated_fn(*args, **kwargs)
```
Comment (dpeng817) [inline python_modules/dagster/dagster/core/definitions/partition.py:702]: we _could_ add some sort of validation step, but I'm not sure if that's even necessary / what we get out of it.

**Review by dpeng817 — APPROVED** [review]
LGTM. I guess this should be followed up by docs yea?

**Comment by sryza** [comment]
@dpeng817 good point about the docs - I added that to this PR.

**Review by dpeng817 — APPROVED** [review]
lgtm!

**Pull request merged**